contracts/multihop/src/tests/swap.rs

```diff
@@ -3,158 +3,363 @@ use crate::storage::Swap;
-        admin: admin.clone(),
-        fee_recipient: user.clone(),
-        lp_wasm_hash: lp_wasm_hash.clone(),
-        max_allowed_slippage_bps: 5000,
-        max_allowed_spread_bps: 500,
-        share_token_decimals: 7,
-        swap_fee_bps: 0,
-        token_init_info: first_token_init_info.clone(),
-        stake_init_info: first_stake_init_info,
+    let swap3 = Swap {
+        offer_asset: token3.address.clone(),
+        ask_asset: token4.address.clone(),
     };
 
-    let second_token_init_info = TokenInitInfo {
-        token_wasm_hash: install_token_wasm(&env),
-        token_a: tokens[1].address.clone(),
-        token_b: tokens[2].address.clone(),
-    };
-    let second_stake_init_info = StakeInitInfo {
-        stake_wasm_hash: install_stake_wasm(&env),
-        min_bond: 5i128,
-        max_distributions: 5u32,
-        min_reward: 2i128,
+    let operations = vec![&env, swap1, swap2, swap3];
+
+    multihop.swap(&recipient, &operations, &50i128);
+
+    // 5. check if it goes according to plan
+    assert_eq!(token1.balance(&recipient), 0i128);
+    assert_eq!(token4.balance(&recipient), 50i128);
+}
+
+#[test]
+fn swap_single_pool_no_fees() {
+    let env = Env::default();
+    let admin = Address::random(&env);
+
+    env.mock_all_auths();
+    env.budget().reset_unlimited();
+
+    let token1 = deploy_and_mint_tokens(&env, &admin, 1_001_000i128);
+    let token2 = deploy_and_mint_tokens(&env, &admin, 1_001_000i128);
+
+    // 1. deploy factory
+    let factory_client = deploy_and_initialize_factory(&env, admin.clone());
+
+    deploy_and_initialize_lp(
+        &env,
+        &factory_client,
+        admin.clone(),
+        token1.address.clone(),
+        1_000_000,
+        token2.address.clone(),
+        1_000_000,
+        None,
+    );
+
+    // 4. swap with multihop
+    let multihop = deploy_multihop_contract(&env, admin.clone(), &factory_client.address);
+    let recipient = Address::random(&env);
+    token1.mint(&recipient, &50i128); // mints 50 token0 to recipient
+    assert_eq!(token1.balance(&recipient), 50i128);
+    assert_eq!(token2.balance(&recipient), 0i128);
+
+    let swap1 = Swap {
+        offer_asset: token1.address.clone(),
+        ask_asset: token2.address.clone(),
     };
 
-    let second_lp_init_info = LiquidityPoolInitInfo {
-        admin: admin.clone(),
-        fee_recipient: user.clone(),
-        lp_wasm_hash: lp_wasm_hash.clone(),
-        max_allowed_slippage_bps: 4_000,
-        max_allowed_spread_bps: 400,
-        share_token_decimals: 6,
-        swap_fee_bps: 0,
-        token_init_info: second_token_init_info,
-        stake_init_info: second_stake_init_info,
+    let operations = vec![&env, swap1];
+
+    multihop.swap(&recipient, &operations, &1);
+
+    // 5. check if it goes according to plan
+    assert_eq!(token1.balance(&recipient), 49i128); // -1 token0
+    assert_eq!(token2.balance(&recipient), 1i128); // +1 token1
```

Comment:
Increase those numbers, swapping of 1 doesn't really show anything.
```suggestion
    multihop.swap(&recipient, &operations, &50);

    // 5. check if it goes according to plan
    assert_eq!(token1.balance(&recipient), 0i128);
    assert_eq!(token2.balance(&recipient), 50i128);
```